extensions/src/main/java/org/apache/falcon/extensions/mirroring/hdfs/HdfsMirroringExtension.java

```diff
@@ -81,17 +83,46 @@ public Properties getAdditionalProperties(final Properties extensionProperties)
         if (StringUtils.isNotBlank(srcPaths)) {
             String[] paths = srcPaths.split(COMMA_SEPARATOR);
 
+            URI pathUri;
             for (String path : paths) {
-                StringBuilder srcpath = new StringBuilder(srcClusterEndPoint);
-                srcpath.append(path.trim());
-                srcpath.append(COMMA_SEPARATOR);
-                absoluteSrcPaths.append(srcpath);
+                try {
+                    pathUri = new URI(path.trim());
+                } catch (URISyntaxException e) {
+                    throw new FalconException(e);
+                }
+                String authority = pathUri.getAuthority();
+                if (authority == null) {
+                    StringBuilder srcpath = new StringBuilder(srcClusterEndPoint);
+                    srcpath.append(path.trim());
+                    srcpath.append(COMMA_SEPARATOR);
+                    absoluteSrcPaths.append(srcpath);
+                } else {
+                    StringBuilder srcpath = new StringBuilder();
+                    srcpath.append(path.trim());
```

Comment:
Minor nit : There is repetition of code here. It might be cleaner to do the following, 
`
 if (authority == null) {
     StringBuilder srcpath = new StringBuilder(srcClusterEndPoint);
} else {
     StringBuilder srcpath = new StringBuilder();
}
srcpath.append(path.trim());
srcpath.append(COMMA_SEPARATOR);
absoluteSrcPaths.append(srcpath);
`